agent/tcs/handler/handler.go

```diff
@@ -95,8 +95,17 @@ func startSession(url string, region string, credentialProvider credentials.AWSC
 
 	defer client.Close()
 
-	client.AddRequestHandler(heartbeatHandler(client))
-	client.AddRequestHandler(ackPublishMetricHandler)
+	// start a timer and listens for tcs heartbeats/acks. The timer is reset when
+	// we receive a heartbeat from the server or when a publish metrics message
+	// is acked.
+	timer := time.AfterFunc(utils.AddJitter(heartbeatTimeout, heartbeatJitter), func() {
```

Comment:
I think you want to throw in a defer timer.Stop to prevent logging the below even after returning from this function (which will only happen if the client is closed anyways).

I think the acs client has this problem as well.